common/projectconfig/projectconfig.go

```diff
@@ -37,15 +37,28 @@ type Config struct {
 	FTLMinVersion string                      `toml:"ftl-min-version"`
 }
 
+// Root directory of the project.
+func (c Config) Root() string {
+	if !filepath.IsAbs(c.Path) {
+		panic(fmt.Errorf("project config path must be absolute: %s", c.Path))
+	}
+	return filepath.Dir(c.Path)
+}
+
 // AbsModuleDirs returns the absolute path for the module-dirs field from the ftl-project.toml, unless
 // that is not defined, in which case it defaults to the root directory.
 func (c Config) AbsModuleDirs() []string {
 	if len(c.ModuleDirs) == 0 {
 		return []string{filepath.Dir(c.Path)}
 	}
+	root := c.Root()
 	absDirs := make([]string, len(c.ModuleDirs))
 	for i, dir := range c.ModuleDirs {
-		absDirs[i], _ = filepath.Abs(dir)
+		cleaned := filepath.Clean(filepath.Join(root, dir))
+		if !strings.HasPrefix(cleaned, root) {
```

Comment:
Does this run into similar issues that we had in `go-runtime/compile` where `.../example2` counts as within `.../example` because of the `HasPrefix` call? Or does root always end in `/`?